types/react-redux/index.d.ts

```diff
@@ -19,6 +19,10 @@ interface ComponentDecorator<TStateProps, TDispatchProps, TOwnProps> {
     (component: Component<TStateProps & TDispatchProps & TOwnProps>): ComponentClass<TOwnProps>;
 }
 
+interface MergedComponentDecorator<TOwnProps, TMergedProps> {
```

Comment:
`ComponentDecorator` is just a special case of `MergedComponentDecorator`. I think they should be merged just because there's already a ton of interfaces and indirection in these typings already.